### PR TITLE
stackrox: test golang upgrade to 1.17.12

### PR DESCRIPTION
**Pull request by rukletsov**
As per guidelines [here](https://github.com/stackrox/stackrox/blob/917c69412556b2d647c21be1bb2f2b33cf200ed8/.openshift-ci/Dockerfile.build_root#L7), this is WIP PR to test if the https://github.com/stackrox/stackrox/pull/2762 which updates our CI image.

Look at individual commits to better understand what has changed (clone and modify operations are in separate commits).

/hold
/uncc

### CODE REVIEW

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-alexr-golang-bump.yaml [hunk 1]
```diff
@@ -0,0 +1,581 @@
...
+        source_path: /go/src/github.com/stackrox/stackrox/image/rhel/.
+  to: main-rcd
+promotion:
+  excluded_images:
+  - main-bundle
+  - central-db-bundle
+  - main-rcd-bundle
+  - operator-artifacts
+  namespace: stackrox
+  tag: latest
```
Comment (gavin-stackrox) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-alexr-golang-bump.yaml:133]:
```suggestion
  tag: alexr-hacks
```

File: ci-operator/config/stackrox/stackrox/stackrox-stackrox-alexr-golang-bump__stackrox_branding.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      paths:
+      - destination_dir: .
+        source_path: /go/src/github.com/stackrox/stackrox/image/postgres/.
+  to: central-db
+promotion:
+  excluded_images:
+  - main-bundle
+  - central-db-bundle
+  namespace: stackrox
+  tag: latest-stackrox-branding
```
Comment (gavin-stackrox) [inline ci-operator/config/stackrox/stackrox/stackrox-stackrox-alexr-golang-bump__stackrox_branding.yaml:60]:
```suggestion
  tag: alexr-hacks-stackrox-branding
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]
those two changes + `make update` + commit + push :sigh: should get the rehearse jobs going.

**Comment by rukletsov** [comment]
2 failed jobs are due to failed tests, the CI builder image clearly works. Closing this PR and proceeding with https://github.com/stackrox/stackrox/pull/2762.

**Pull request closed without merging**